### PR TITLE
feat(core): new method getFacilitiesByAttributeWithAttributes

### PR DESCRIPTION
**Pull request by xflord**
* implemented new method getFacilitiesByAttributeWithAttributes that searches for facilities by attribute and returns them along with additional chosen attributes
* added new class FacilityWithAttributes

### CODE REVIEW

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntryIntegrationTest.java [hunk 1]
```diff
@@ -235,6 +236,40 @@ public void getFacilitiesByAttribute() throws Exception {
...
+			attrDef = perun.getAttributesManagerBl().createAttribute(sess, attrDef);
+		}
+
+		attr = new Attribute(attrDef);
+		attr.setValue("valueeeee");
+
+		// Set the attribute to the facility
+		perun.getAttributesManagerBl().setAttribute(sess, facility, attr);
+
+		System.out.println(perun.getAttributesManager().getAttributes(sess, facility));
```
Comment (Johaney-s) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntryIntegrationTest.java:265]: probably debugging leftover

File: perun-utils/rpc-methods-javadoc-generator/parseRpcMethods.pl [hunk 2]
```diff
@@ -216,6 +216,10 @@
 $objectExamples{"List&lt;EnrichedResource&gt;"} = $listPrepend . $objectExamples{"EnrichedResource"} . $listAppend;
 $objectExamples{"List<EnrichedResource>"} = $objectExamples{"List&lt;EnrichedResource&gt;"};
 
+$objectExamples{"FacilityWithAttributes"} = "{ \"facility\" : " .$objectExamples{"Facility"} . " , \"attributes\" : " .$objectExamples{"List&lt;Attribute&gt;"} . " }";
+$objectExamples{"List&lt;FacilityWithAttributes&gt;"} = $listPrepend . $objectExamples{"FacilityWithAttributes"} . $listAppend;
+$objectExamples{"List<FacilityWithAttributes>"} = $objectExamples{"List&lt;FacilityWithAttributes&gt;&gt;"};
```
Comment (Johaney-s) [inline perun-utils/rpc-methods-javadoc-generator/parseRpcMethods.pl:221]:
```suggestion
$objectExamples{"List<FacilityWithAttributes>"} = $objectExamples{"List&lt;FacilityWithAttributes&gt;"};
```

File: perun-core/src/test/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntryIntegrationTest.java [hunk 3]
```diff
@@ -235,6 +236,40 @@ public void getFacilitiesByAttribute() throws Exception {
 		assertTrue("results must contain user", facilitiesManagerEntry.getFacilitiesByAttribute(sess, attributeName, "value").contains(facility));
 	}
 
+	@Test
+	public void getFacilitiesByAttributeWithAttributes() throws Exception {
+		System.out.println(CLASS_NAME + "getFacilitiesByAttributeWithAttributes");
+
+		// Check if the attribute already exists
+		Attribute attr;
```
Comment (Johaney-s) [inline perun-core/src/test/java/cz/metacentrum/perun/core/entry/FacilitiesManagerEntryIntegrationTest.java:244]: would be nice to reuse some setUp method, for example `Attribute attr = setUpAttribute1(); `, and move the try-catch block to the setUp method instead

**Review by Johaney-s — CHANGES_REQUESTED** [review]
seems to work fine, only a few suggestions

**Review by Johaney-s — APPROVED** [review]

**Review by sarkapalkovicova — APPROVED** [review]

**Pull request merged**